mutualcap_button_wizard.py

```diff
@@ -0,0 +1,117 @@
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+#  MA 02110-1301, USA.
+#
+
+import FootprintWizardBase
+import pcbnew
+from pcbnew import *
+
+class MutualcapButtonWizard(FootprintWizardBase.FootprintWizard):
+
+    def GetName(self):
+        """
+        Return footprint name.
+        This is specific to each footprint class, you need to implement this
+        """
+        return "Mutualcap Touch Button"
+
+    def GetDescription(self):
+        """
+        Return footprint description.
+        This is specific to each footprint class, you need to implement this
+        """
+        return "Wizard for Mutual Capacitance Touch Buttons."
+
+    def GetValue(self):
+        return "TouchButton-{width:g}x{height:g}mm".format(
+            width = pcbnew.ToMM(self.pads["Width"]),
+            height= pcbnew.ToMM(self.pads["Height"])
+            )
+
+    def GenerateParameterList(self):
+        #self.AddParam("Pads", "panel_thickness",self.uMM)
+        self.AddParam("Pads", "T (Panel thickness)", self.uMM, 1.6)
```

Comment:
The values 'T' and 'Y width' are hard to parse.  I see they are used that way by Atmel, but let's get a better description.  Maybe "Surrounding Width"?

Reply:
How about pt (panel thickness) and re and de for Drive/Receive electrode? Drive Electrode (currently X) is the thick surrounding one while Receive Electrode (currently Y) would be the inner thin one. 
Just calling it "Surrounding Width" i think would be a bit unclear as to what it is because the Drive Electrode "Finger" thickness also depends on that parameter.

Reply:
I'd prefer to stay away from functional representations.  Instead, let's keep the graphical representations.  Maybe "Inner width" and "Outer width"?  Thickness tends to refer to the z-direction so is confusing in this context.

Reply:
I see that inner is changed but there is still "Panel Thickness".  This is an uncommon term.  It would be good to rename it.  If you dislike "Outer width", you might call it "Frame Width".

Reply:
I think there is a misunderstanding. The "Panel Thickness" refers to the thickness (z-direction) of whatever graphics panel or protection layer you put physically above the touchpad on the PCB (For example a acrylic sheet as graphical overlay or protection so that you dont touch the pcb directly). The "Outer width" is derived from the other parameters and can not be manually entered as a number.

Atmel calls it a front panel, Cypress (where this picture is from) calls it Overlay
![article-2016december-simplifying-capacitive-touch-fig1](https://www.digikey.be/-/media/Images/Article%20Library/TechZone%20Articles/2016/December/Simplifying%20Capacitive%20Touch%20Sensor%20Design%20Cypress%20CY8CKIT/article-2016december-simplifying-capacitive-touch-fig1.jpg)

The internal variables were already changed to have i and o prefixes for inner and outer.

Reply:
Thank you for that additional information.  I wasn't away of the relationship.

That being said, the border width or frame width is the same value as the overlay thickness in this recommended implementation.  A user would not be expected to know the relationship but they would be concerned about the width of the copper that is generated.  You could explain the relationship in the wizard description text but I'd prefer that the parameter be focused on the copper that is generated.

Reply:
okay, parameter strings have been changed to inner and outer